src/views/payment-sheet-views/card-view.js

```diff
@@ -74,40 +91,65 @@ CardView.prototype.initialize = function () {
   }.bind(this));
 };
 
-CardView.prototype._setupCardholderName = function (cardholderNameField) {
-  var cardholderNameOptions = this.model.merchantConfiguration.card && this.model.merchantConfiguration.card.cardholderName;
-  var cardholderNameContainer = cardholderNameField.querySelector('.braintree-form__hosted-field');
+CardView.prototype._setupExtraInput = function (extraInput) {
+  var self = this;
+  var fieldNameKebab = camelCaseToKebabCase(extraInput.fieldName);
+  var field = this.getElementById(fieldNameKebab + '-field-group');
+  var input = field.querySelector('input');
+  var nameContainer = field.querySelector('.braintree-form__hosted-field');
 
-  this.cardholderNameInput.addEventListener('keyup', function () {
-    var hasContent = this.cardholderNameInput.value.length > 0;
+  input.addEventListener('keyup', function () {
+    var valid = true;
+
+    extraInput.validations.forEach(function (inputValidation) {
+      valid = valid && inputValidation.isValid(input.value);
+    });
 
-    classlist.toggle(cardholderNameContainer, 'braintree-form__field--valid', hasContent);
+    classlist.toggle(nameContainer, 'braintree-form__field--valid', valid);
 
-    if (!cardholderNameOptions.required) {
+    if (!extraInput.required) {
       return;
     }
 
-    if (hasContent) {
-      classlist.remove(cardholderNameField, 'braintree-form__field-group--has-error');
+    if (valid) {
+      self.hideFieldError(extraInput.fieldName);
     }
 
-    this._sendRequestableEvent();
-  }.bind(this), false);
+    self._sendRequestableEvent();
+  }, false);
 
-  if (cardholderNameOptions.required) {
-    this.cardholderNameInput.addEventListener('blur', function () {
-      // the active element inside the blur event is the docuemnt.body
+  if (extraInput.required) {
+    input.addEventListener('blur', function () {
+      // the active element inside the blur event is the document.body
       // by taking it out of the event loop, we can detect the new
       // active element (hosted field or other card view element)
       setTimeout(function () {
-        if (isCardViewElement() && this.cardholderNameInput.value.length === 0) {
-          classlist.add(cardholderNameField, 'braintree-form__field-group--has-error');
+        var hasFieldError = false;
+
+        if (isCardViewElement()) {
+          if (input.value.length == 0) {
+              hasFieldError = true;
+              self.showFieldError(extraInput.fieldName, extraInput.requiredError);
+          }
+
+          extraInput.validations.forEach(function (validation) {
+            if (!hasFieldError && !validation.isValid(input.value)) {
+              hasFieldError = true;
+              self.showFieldError(extraInput.fieldName, validation.error);
+            }
+          });
         }
-      }.bind(this), 0);
-    }.bind(this), false);
+      }, 0);
+    }, false);
   }
 };
 
+CardView.prototype._removeExtraInput = function (extraInput) {
```

Comment:
I think we're following the pattern of how the hosted fields stuff is set up, where we remove the postal code input if that challenge is not required.

I'd be fine with programmatically inserting it though. Should probably do the same (in a future pr) for the Hosted Fields inputs.

Reply:
Yeah I was just thinking because they're "extra" it's odd to include them by default, but I think it'd be the cleanest to just programatically generate all the fields.